src/time.c

```diff
@@ -99,8 +100,9 @@ int set_unix_time_zone (const char *time_zone)
        debug_error("set_unix_time_zone() error, TZ = %s\n", time_zone);
    }
 
+   set_gmtoff(mt->tm_gmtoff);
```

Comment:
I don't think this approach will work because you're setting the gmt offset once & only when the TZ is first set.  If it happens to be during daylight savings hours, the offset will always be offset that amount.  The same is true during non-daylight savings hours.